app/src/main/java/com/example/ohthmhyh/fragments/UserFragment.java

```diff
@@ -89,34 +73,104 @@ public void onClick(View view) {
+        EditText searchFriendsET = view.findViewById(R.id.friend_Search_ET);
+        Button searchBtn = view.findViewById(R.id.send_request_btn);
+
+        // set the views
+        usernameTV.setText(user.getUsername());
+        userBioTV.setText(user.getBio());
+        emptyRequestTV.setText("Looks like you're all caught up!");
+        emptyFriendsTV.setText("You're not following anyone!");
+
+        // fill the friends list view
+        friendsLV.setEmptyView(emptyFriendsTV);
+        FriendsListAdaptor friendsAdaptor = new FriendsListAdaptor(getActivity(),
+                R.layout.item_friend, user.getFriendList());
+        friendsLV.setAdapter(friendsAdaptor);
+
+        // fill the friend request list view
+        requestLV.setEmptyView(emptyRequestTV);
+        FriendRequestListAdaptor FRAdaptor = new FriendRequestListAdaptor(getActivity(),
+                R.layout.item_friend_request, user.getFriendRequests());
+        FRAdaptor.setCustomButtonListener(new FriendRequestListAdaptor.buttonListener() {
+            @Override
+            public void onAcceptClickListener(int position) {
+                Toast.makeText(getContext(), "ACCEPT "+position, Toast.LENGTH_SHORT).show();
+                user.acceptFriendRequest(position);
+                friendsAdaptor.notifyDataSetChanged();
+                FRAdaptor.notifyDataSetChanged();
+            }
 
-        Button editProfileButton = view.findViewById(R.id.user_editprofile);
-        editProfileButton.setOnClickListener((v) -> {
-            Intent editProfile = new Intent(getActivity(), EditProfileActivity.class);
-            editProfile.putExtra("NAME", user.getDisplayName());
-            //editProfile.putExtra("USERNAME", user.get)
-            getActivity().startActivity(editProfile);
+            @Override
+            public void onDeclineClickListener(int position) {
+                Toast.makeText(getContext(), "DECLINE "+position, Toast.LENGTH_SHORT).show();
+                user.denyFriendRequest(position);
+                FRAdaptor.notifyDataSetChanged();
+            }
         });
+        requestLV.setAdapter(FRAdaptor);
 
+        // send friend request when the button is pressed
+        searchBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String username = searchFriendsET.getText().toString();
+                searchFriendsET.getText().clear();
+
+                // make sure the username is valid
+                DatabaseAdapter.checkUsernameExists(username,
+                        new DatabaseAdapter.UsernameCheckCallback() {
+                            @Override
+                            public void onUsernameCheckCallback(boolean usernameExists) {
+                                if(usernameExists){
+                                    // send the friend request
+                                    user.sendFriendRequest(username);
```

Comment:
I'm not sure how easy it would be, but I wonder if this method can return a boolean indicating whether the friend request was sent or not?

Right now, when I try to send a friend request to myself or a user already in my following list, I still get a Toast message saying that the friend request was sent.

Reply:
Definitely doable, but it's going to require more callbacks. Up to you what you want to do.